### ISSUE TITLE
a11y: Add token to support disabled text

### ISSUE THREAD

**Issue by blackfalcon** [AlaskaAirlines/AuroDesignTokens#195]
### Is your feature request related to a problem? Please describe.

In reviewing Flight Search for accessibility issues, I came across the UI attached to this issue. 

<img width="1310" alt="Screenshot 2024-11-27 at 10 55 48 AM" src="https://github.com/user-attachments/assets/63365877-a39a-4991-941e-efb1b7697325">

The CSS is written as 

```css
.valuetile.svelte-1unl1vz .valuetile-container .price-callout-container .disabled-text.svelte-1unl1vz {
    color: var(--ds-color-border-disabled-default);
    font-size: var(--ds-text-body-size-sm);
}
```

The code `color: var(--ds-color-border-disabled-default);` is the focus of this issue. 

1. `--ds-color-border-disabled-default` is not the right token to be used for text
2. `--ds-color-border-disabled-default` is a deprecated token
3. There is no relative semantically correct token to request for replacement and is suspect to failed theming UI

### Describe the solution you'd like

New tokens specifically for disabled text is requested. 

```css
var(--ds-color-text-disabled-default),
var(--ds-color-text-disabled-inverse),
```

According to WCAG 2.1 - Success Criterion 1.4.11: Non-Text Contrast, this requires a minimum 3:1 contrast ratio for graphical elements and states that it applies to "inactive user interface components," such as disabled text.

The closeted existing Auro token would be `var(--ds-color-base-gray-500)` with a hex value of `#959595` which barely meets the 3:1 contrast requirement (2.99:1) but can be considered an acceptable option for disabled text.

For the inverse, `var(--ds-color-base-gray-600)` with a hex value of `#7e7e7e` would meet the 3.1 color contrast ratio 3.1 requirements (3.5:1) using the `var(--ds-color-background-primary-100-inverse)` token with hex value of `#0e2b4f` as an inverse background.

I'd also recommend that the token `var(--ds-color-text-ui-disabled-default)` be updated to alias `color.base.gray.500.value` to support the `#959595` hex value that meets the 3:1 contrast requirement.

### Describe alternatives you've considered

The cost of doing nothing is the following; endangerment of not providing a semantic token for disabled text which will result in inconsistent theme updates going forward and the possibility of disengaging users with low-vision impairments and/or color blindness issues. 

### Additional context

Related issue for flight search team

- https://github.com/Alaska-ECommerce/flightsearch-lem-svelte/issues/2271

### Exit criteria

This issue can be closed once the requested tokens have been made available for consumption by UI development teams.

**Comment by jason-capsule42** [comment]
@blackfalcon we currently already have these two tokens for this purpose defined:

```css
--ds-color-text-ui-disabled-default: #adadad;
--ds-color-text-ui-disabled-inverse: #7e7e7e;
```

@nklansford We need to make sure we consider this as part of the semantic token conversations that are ongoing.

**Comment by blackfalcon** [comment]
Hey @jason-capsule42 

I did see those tokens in there, but I am unsure as to the rules for using `ui` tokens. These were initially intended to be used for 'interactive' elements and this use case is for non-interactive text that is to appear as disabled within the search experience. 

Also, there is a suggested change to a token to use `"value": "{color.base.gray.500.value}",` as the `disabled-default` versus the `"value": "{color.base.gray.400.value}",` in order to meet a 3:1 contrast ratio that is documented in this issue. 

If the decision is to retain only the `--ds-color-text-ui-disabled-default/inverse` tokens, I am ok with that, but I would strongly recommend changing the token alias reference.

**Comment by jason-capsule42** [comment]
Totally agree with where your head is at on this @blackfalcon. 

There is a conversation being headed up by @nklansford assessing our semantic naming convention. As more teams are being required to use the tokens to support dual brand theming the complexity of the naming solution in place and the confusion arising from that is showing itself. There is movement to simplify the semantic token names, make them much more obvious about which to use and where and hopefully prevent issues exactly like this in the future.

**Comment by jason-capsule42** [comment]
I am going to move this to BLOCKED until the new naming convention is resolved and then I'll come back to this ticket.

**Comment by blackfalcon** [comment]
A message was sent to @megrenn for follow-up and information on how to resolve.

**Comment by NickingAS** [comment]
Hi All. 

it seems there was a misquote of the WCAG Guidelines for contrast minimums. Contrast minimums do not apply to inactive or disabled components and text. 

Please see the applicable call outs below and feel free to check the [WCAG guidelines for contrast minimums here](https://www.w3.org/TR/WCAG21/#contrast-minimum).



Success Criterion 1.4.3 Contrast (Minimum)

(Level AA)
The visual presentation of text and images of text has a contrast ratio of at least 4.5:1, **except for the following**:
_Large Text_
Large-scale text and images of large-scale text have a contrast ratio of at least 3:1;
**_Incidental_
Text or images of text that are part of an inactive user interface component**, that are pure decoration, that are not visible to anyone, or that are part of a picture that contains significant other visual content, have no contrast requirement.
_Logotypes_
Text that is part of a logo or brand name has no contrast requirement.



Success Criterion 1.4.6 Contrast (Enhanced)

(Level AAA)
The visual presentation of text and images of text has a contrast ratio of at least 7:1, **except for the following**:
_Large Text_
Large-scale text and images of large-scale text have a contrast ratio of at least 4.5:1;
**_Incidental_
Text or images of text that are part of an inactive user interface component**, that are pure decoration, that are not visible to anyone, or that are part of a picture that contains significant other visual content, have no contrast requirement.
_Logotypes_
Text that is part of a logo or brand name has no contrast requirement.



Success Criterion 1.4.11 Non-text Contrast

(Level AA)
The visual presentation of the following have a contrast ratio of at least 3:1 against adjacent color(s):
_User Interface Components_
Visual information required to identify user interface components and states, **except for inactive components** or where the appearance of the component is determined by the user agent and not modified by the author;
_Graphical Objects_
Parts of graphics required to understand the content, except when a particular presentation of graphics is essential to the information being conveyed.

Since the associated color tokens fall within these guidelines, from a UI and WCAG Contrast Compliance POV no change is needed to the disabled colors.

**Comment by blackfalcon** [comment]
This will be the official position on this issue from now on. 

> Disabled text (part of inactive UI components) is exempt from minimum contrast requirements under WCAG 2.1 Success Criterion 1.4.3. However, to support better usability and legibility, especially for users with low vision, many accessibility professionals still recommend a contrast ratio of at least 3:1 or better, even for disabled states—though it's not strictly required.
